### DISCUSSION TITLE
Debounce implementation

### DISCUSSION THREAD

**Discussion by cody-elhard** [mhenrixon/sidekiq-unique-jobs#760] [General]
Hi everyone,

I would like to debounce a job that updates user details at most once per day.

Debounce meaning it should run at the very end after the lock is released. We want it to run at the very end to get the latest details. There is a chance the user will be out of the software for some time, which is why a throttle mechanism is not ideal.

Are locks capable of this or should I be using a different solution?

**Comment by cody-elhard** [comment]
`until_expired` gave me exactly what I needed

**Reply by mhenrixon** [reply]
The names and such are possibly not ideal. 

I like debounce for that job but I wasn't aware of the term before a few months ago.
